app/assets/javascripts/actions/story.js

```diff
@@ -26,24 +26,34 @@ export const editStory = (id, newAttributes) => ({
   newAttributes
 });
 
+export const setLoading = (id) => ({
+  type: actionTypes.SET_LOADING,
+  id
+});
+
 export const updateStory = (story, projectId) =>
   (dispatch, getState, { Story }) => {
     if (story._editing._isDirty) {
+      dispatch(setLoading(story.id))
       return Story.update(story._editing, projectId)
         .then((story) => {
           dispatch(updateStorySuccess(story));
-        });
+        })
+        .then(() => setLoading(story.id));
     }
     return dispatch(toggleStory(story.id));
   };
 
 export const deleteStory = (storyId, projectId) =>
-  (dispatch, getState, { Story }) =>
-    Story.deleteStory(storyId, projectId)
+  (dispatch, getState, { Story }) => {
+  dispatch(setLoading(storyId))
+    return Story.deleteStory(storyId, projectId)
       .then(() => dispatch(deleteStorySuccess(storyId)))
+      .then(() => dispatch(setLoading(storyId)))
```

Comment:
@talyssonoc here we need to put this dispatch inside the catch as well? in case `dispatch(deleteStory)` fails?